frontend/angular/style-guide.md

```diff
@@ -0,0 +1,972 @@
+      const { offices, woloxers, departments, techs } = data;
+      this.offices = offices;
+      this.woloxers =  woloxers;
+      this.departments = departments;
+      this.techs = techs;
+    });
+  heroes: Hero[] = [];
+
+  constructor(private AppService: AppService) {}
+
+  ngOnInit() {
+    this.AppService.getHeroes().then((heroes) => (this.heroes = heroes));
+  }
+}
+```
+
+In `routing.module` add a resolver
+```ts
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
+import { HomeComponent } from './home.component';
+import { ResolverService } from './services/resolver.service' // new line
+const routes: Routes = [
+  {
+    path: '',
+    component: HomeComponent,
+    children: [
+      {
+        path: '**',
+        redirectTo: ''
+      }
+    ],
+    resolve: { resolved: ResolverService }, // new line
+    runGuardsAndResolvers: 'always' // new line
+  }
+];
+
+@NgModule({
+  imports: [RouterModule.forChild(routes)],
+  exports: [RouterModule]
+})
+export class HomeRoutingModule { }
+
+```
+
+And finally provider service on Home.module
+```ts
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+import { HomeComponent } from './home.component';
+import { HomeRoutingModule } from './home-routing.module';
+import { ResolverService } from './services/resolver.service'; // new line
+
+@NgModule({
+  declarations: [ HomeComponent ],
+  imports: [
+    CommonModule,
+    HomeRoutingModule
+  ],
+  providers: [ResolverService] // new line
+})
+export class HomeModule {}
+```
+
+## Access modifiers
+
+- If your properties are being used in the template, then they cannot be privates.
+
+```ts
+@Component({
+  selector: 'app-cool',
+  template: `<div>I'm {{ name }}</div>`,
+})
+export class AppComponent {
+  name = 'wolox';
+}
+```
+
+**Avoid**
+
+```ts
+@Component({
+  selector: 'app-cool',
+  template: `<div>I'm {{ name }}</div>`,
+})
+export class AppComponent {
+  private name = 'wolox';
+}
+```
+
+> That's applied conversely, namely, if you don't use the properties in the template, they **must** be `private`. When your app is in production mode, Ahead-of-Time compilation won't allow it's visibility in the template.
+
+- If your property is not mutated at all in your class, then it **must be** `readonly`.
+
+## Properties order
+
+- decorators
+  - inputs
+  - outputs
+  - others
+- public properties
+- public readonly properties
+- private properties
+- private readonly properties
+- setter and getter properties (accessors)
+  - public
+  - private
+
+```ts
+@Component({
+  selector: 'app-cool',
+  template: `<div>I'm a nice App</div>`,
+})
+export class AppComponent {
+  @Input() foo;
+  @Ouput() fooChange = new EventEmiter();
+  isCool = true;
+  private name = 'wolox';
+  readonly place = 'world';
+
+  get woo() {
+    return this.anyValue;
+  }
+
+  set woo(value) {
+    this.anyValue = value;
+  }
+
+  constructor() {}
+}
```

Comment:
```suggestion
@Component({
  selector: 'app-cool',
  template: `<div>I'm a nice App</div>`,
})
export class AppComponent {
  @Input() foo;
  @Ouput() fooChange = new EventEmiter();
  isCool = true;
  private name = 'wolox';
  readonly place = 'world';

  constructor() {}
 
   lifeCycles() {} 
 
  get woo() {
    return this.anyValue;
  }

  set woo(value) {
    this.anyValue = value;
  }

otherMethods() {}
}
```

Reply:
I like this structure order. What do you think? @CrisVega08

Reply:
So... will we consider accessor properties like methods by a visual concept?

Reply:
I think methods and accessor are different and Angular interprets these in different ways, so I prefer the way we defined

Reply:
OK @FrankIglesias  and @LucasZibell , I would like know why do you prefer after lifecycles ?

Reply:
I think that what Frank proposes allows you to have all the logic in the same place. Allowing you to rappidly navigate through the code. Maybe in the example given above the logic is very simple as it is just a return or an assign. But for example, in cases more complex like this `this._name = (name && name.trim()) || '<no name set>';` given in the Angular [Doc](https://angular.io/guide/component-interaction) it allows you to have all the logic compress in near lines of the code.
For the question of why after or before the lyfecycles i think it is a nice order having into account the life of a component. You can have it like this:
```
Component {
-> First the state
-> Contructor
--- Now here will start the logic of the component ---
-> Lifecycle // This will allow you to see how the component will react after it is contructed, what will be its behavior when it ts mounted/created/destroyed
-> Getters and setters // Here you will have the logic according to the access or setting of a specifici state
-> Other methods // Logic specific of the component, will be the methods to which it will respond
}
``` 
I don't agree with Beto in saying it will be visually a method, as all the getters and setters start with `get/set` making a very clear difference between an accessor and a method. The question now could be if this accessors go before or after the lifecylce, as i said and explain above in my point of view it will be better to have them afterwards, keeping all the lifecycle together with the constructor.

Reply:
Ok I understood you idea, I prefer before lifecycle because:

1. When you set or reset data-bound input properties angular responds with lifecycle ngOnChanges.
2. I see getters into middle between properties and methods